src/workerd/jsg/modules.c++

```diff
@@ -587,7 +605,16 @@ v8::Local<v8::Value> NodeJsModuleContext::require(jsg::Lock& js, kj::String spec
   auto modulesForResolveCallback = jsg::getModulesForResolveCallback(js.v8Isolate);
   KJ_REQUIRE(modulesForResolveCallback != nullptr, "didn't expect resolveCallback() now");
 
-  kj::Path targetPath = path.parent().eval(specifier);
+  kj::Path targetPath = ([&] {
+    // If the specifier begins with one of our known prefixes, let's not resolve
+    // it against the referrer.
+    if (specifier.startsWith("node:") ||
+        specifier.startsWith("cloudflare:") ||
+        specifier.startsWith("workerd:")) {
```

Comment:
Non-blocking: would be nice to refactor this out into a separate function to avoid duplication.

Reply:
As a follow on to the module registry refactor I expect much of this to be cleaned up anyway so I'm not that concerned about the duplication right now.